api/net/router.hpp

```diff
@@ -168,10 +168,18 @@ namespace net {
      * @todo : Optimize!
      **/
     Route<IPV>* get_cheapest_route(typename IPV::addr dest) {
-      Routing_table all = get_all_routes(dest);
-      std::sort(all.begin(), all.end());
-      if (not all.empty()) return &all.front();
```

Comment:
Yea, returning a pointer to a temporary here, since get_all_routes creates a vector of fresh copies, which we then sort. Good catch. Another way of fixing it is to not return a raw pointer, but rather`std::optional<Route<IPV>>`, e.g. 

```c++
    std::optional<Route<IPV>> get_cheapest_route(typename IPV::addr dest) {
      Routing_table all = get_all_routes(dest);
      std::sort(all.begin(), all.end());
      if (not all.empty()) return all.front();
      return {};
    };
```
But your loop both avoids the copy and the issue of returning a temporary. Great!